### ISSUE TITLE
Secret: byte[] secret confusion

### ISSUE THREAD

**Issue by anibyl** [BastiaanJansen/otp-java#83]
This ticket is merely a question to start a discussion.

Currently, the library uses the following constructor for initialization ([link](https://github.com/BastiaanJansen/otp-java/blob/6aeafa048ab646bb2ec87133e72b80aafa85d622/src/main/java/com/bastiaanjansen/otp/HOTPGenerator.java#L199)):
```java
public Builder(final byte[] secret) {
```

A **shared secret** is a **byte sequence**, also commonly represented as a **Base32 string** ([example](https://github.com/google/google-authenticator/wiki/Key-Uri-Format#secret)).

However, the builder constructor consumes *neither of them*, but instead a byte array which is a Base32 secret encoded with a standard charset (`secret.getBytes()`, e.g. [here](https://github.com/BastiaanJansen/otp-java/blob/6aeafa048ab646bb2ec87133e72b80aafa85d622/src/test/java/com/bastiaanjansen/otp/TOTPGeneratorTest.java#L77)):
```java
private final static String secret = "vv3kox7uqj4kyakohmzpph3us4cjimh6f3zknb5c2oobq6v2kiyhm27q";
...
TOTPGenerator generator = new TOTPGenerator.Builder(secret.getBytes()).withHOTPGenerator(builder -> {
```

It's unclear why it has to be like that, since the encoded-string secret is later [converted to a correct shared secret](https://github.com/BastiaanJansen/otp-java/blob/6aeafa048ab646bb2ec87133e72b80aafa85d622/src/main/java/com/bastiaanjansen/otp/HOTPGenerator.java#L103) anyway:
```java
private final byte[] secret;
...
byte[] secretBytes = decodeBase32(secret);
```

I recently spent a few hours trying to understand why the library wasn't working when I was passing a correct shared secret to it.

**This is confusing.**

Would that be possible to add a javadoc explaining what should be passed as `byte[] secret`, or a `Builder` constructor consuming `String secret`, or...?

I could create a PR, but first I would like to have a conversation.

**Comment by BastiaanJansen** [comment]
Hmm thanks for your question! You make a good point as there is no specific reason the library works this way. 

It would be appreciated if you make a PR with JavaDoc or another implementation!

**Comment by anibyl** [comment]
Hi @BastiaanJansen, please take a look at #84 :)

**Comment by BastiaanJansen** [comment]
Thanks! This is included in version 2.1.0!
